src/utils/storage.ts

```diff
@@ -250,6 +250,60 @@ export async function processHistory(storage: IStorage, { rewardLock }: { reward
   await updateWalletMetadataFromProcessedTxData(storage, { maxIndexUsed, tokens });
 }
 
+/**
+ * Fetch and save the data of the token set on the storage
+ * @param {IStorage} storage - Storage to save the tokens.
+ * @param {Set<string>} tokens - set of tokens to fetch and save.
+ * @returns {Promise<void>}
+ */
+async function updateTokensData(storage: IStorage, tokens: Set<string>): Promise<void> {
+  async function fetchTokenData(uid: string): Promise<{
+    success: true,
+    name: string,
+    symbol: string,
+  } | { success: false, message: string }> {
+    let retryCount = 0;
+
+    while (retryCount <= 5) {
```

Comment:
Would it make sense to use an exponential backoff for the retries?

Reply:
We can, an idea would be to use:
```js
catch (err) {
  retryCount += 1;
  const delay = 10 * (2 ** retryCount);
  await new Promise(resolve => setTimeout(resolve, delay));
  continue;
}
```

Which would give us 10ms, 20ms, 40ms, 80ms and 160ms of intervals.
We could use another multiplier instead of 10ms to increase the intervals as well.
What do you think?

Reply:
The idea of the exponential backoff would be to give some time for the fullnode to recover. I don't think it would recover in 160ms.

I was thinking of longer delays, like 0.5s, 1s, 2s, 4s, 8s. Unless waiting so long is bad for the use case that needs this response.

Reply:
Done